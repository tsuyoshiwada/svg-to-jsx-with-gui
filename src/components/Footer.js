/* eslint-disable */
import React from 'react';
import styled from 'styled-components';
import { Logo, GitHub } from '../icons/';

const FooterWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  position: fixed;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1000;
  height: 60px;
  background: #4c5275;
  color: #fff;
`;

const FooterCol = styled.div`
  flex: 1 1 33.3333%;
  font-size: 14px;
  padding-left: ${props => props.left ? '20px' : '0'};
  padding-right: ${props => props.right ? '20px' : '0'};
  text-align: ${props => props.right ? 'right' : props.center ? 'center' : 'left'};

  & a {
    text-decoration: none;
    color: inherit;

    &:hover {
      text-decoration: underline;
    }
  }
`;

const FooterTitle = styled.h1`
  margin: 0;
  font-size: 1em;
  font-weight: normal;

  & small {
    display: block;
    margin: 0.2em 0 0;
    font-size: 0.78em;
  }
`;

const GitHubLink = styled.a`
  display: inline-block;
  margin: 0 0 0 20px;
  line-height: 1;
  vertical-align: middle;

  &:hover {
    opacity: 0.6;
  }
`;

const Footer = () => (
  <FooterWrapper>
    <FooterCol left>
      <FooterTitle>
        SVG to JSX with GUI
        <small>Open Source Tool</small>
      </FooterTitle>
    </FooterCol>

    <FooterCol center style={{ lineHeight: 1 }}>
      <Logo
        width={51}
        height={33}
      />
    </FooterCol>

    <FooterCol right>
      Made with <span style={{ fontSize: '1.28em', color: '#fd379a' }}>&#9829;</span> <a href="https://github.com/tsuyoshiwada">@tsuyoshiwada</a>.
      <GitHubLink href="__todo__">
        <GitHub />
      </GitHubLink>
    </FooterCol>
  </FooterWrapper>
);

export default Footer;
