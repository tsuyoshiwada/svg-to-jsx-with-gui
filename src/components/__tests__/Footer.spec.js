import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../Footer';


describe('components#<Footer />', () => {
  test('Should be render', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });
});
