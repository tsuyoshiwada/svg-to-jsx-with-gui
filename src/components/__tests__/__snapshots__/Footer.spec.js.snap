// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components#<Footer /> Should be render 1`] = `
<styled.div>
  <styled.div
    left={true}
  >
    <styled.h1>
      SVG to JSX with GUI
      <small>
        Open Source Tool
      </small>
    </styled.h1>
  </styled.div>
  <styled.div
    center={true}
    style={
      Object {
        "lineHeight": 1,
      }
    }
  >
    <Logo
      height={33}
      width={51}
    />
  </styled.div>
  <styled.div
    right={true}
  >
    Made with 
    <span
      style={
        Object {
          "color": "#fd379a",
          "fontSize": "1.28em",
        }
      }
    >
      â™¥
    </span>
     
    <a
      href="https://github.com/tsuyoshiwada"
    >
      @tsuyoshiwada
    </a>
    .
    <styled.a
      href="__todo__"
    >
      <GitHub />
    </styled.a>
  </styled.div>
</styled.div>
`;
